option optimize_for = SPEED;

package NPrsLogProto;

message TDocument {
  optional string Url = 1;
  optional string DocId = 2;
  optional string OmniTitle = 3;
  optional bytes SerializedFactors = 4;

  optional int64 SourceRelevance = 5;   // L2
  optional int64 MetaRelevance = 6;     // L3
}

message TGroup {
  optional string CategoryName = 1;
  repeated TDocument Documents = 2;
}

message TGrouping {
  repeated TGroup Groups = 1;
}

message TLog {
  optional TGrouping Grouping = 1;
  optional string CompressionMethod = 2 [default = "none"];
  optional bytes CompressedLog = 3;
}
